addons/mail/static/src/core/web/chatter.js

```diff
@@ -194,7 +195,9 @@ export class Chatter extends Component {
                 } else {
                     this.state.showAttachmentLoading = false;
                     this.state.isAttachmentBoxOpened =
-                        this.props.isAttachmentBoxVisibleInitially && this.attachments.length > 0;
+                        (this.props.isAttachmentBoxVisibleInitially &&
+                            this.attachments.length > 0) ||
+                        this.state.wasAttachmentBoxOpened;
```

Comment:
Why do we need another state variable to check whether the attachment box is open when you already have one? Instead, you can do this:

```suggestion
                        this.state.isAttachmentBoxOpened ||
                        (this.props.isAttachmentBoxVisibleInitially &&
                            this.attachments.length > 0)
```

 what do you think ?